rust/src/transformations/make_stable_expr/mod.rs

```diff
@@ -70,3 +70,58 @@ where
         }
     }
 }
+
+#[cfg(test)]
+pub mod polars_test {
+
+    use crate::domains::{AtomDomain, LazyFrameDomain, Margin, SeriesDomain};
+    use crate::error::*;
+    use polars::prelude::*;
+
+    pub fn get_test_data() -> Fallible<(LazyFrameDomain, LazyFrame)> {
+        let lf_domain = LazyFrameDomain::new(vec![
+            SeriesDomain::new("chunk_2_bool", AtomDomain::<bool>::default()),
+            SeriesDomain::new("cycle_5_alpha", AtomDomain::<String>::default()),
+            SeriesDomain::new("const_1f64", AtomDomain::<f64>::default()),
+            SeriesDomain::new("chunk_(..10u32)", AtomDomain::<u32>::default()),
+            SeriesDomain::new("cycle_(..100i32)", AtomDomain::<i32>::default()),
+        ])?
+        .with_margin::<&str>(
+            &[],
+            Margin::new()
+                .with_public_lengths()
+                .with_max_partition_length(1000),
+        )?
+        .with_margin(
+            &["chunk_2_bool"],
+            Margin::new()
+                .with_public_lengths()
+                .with_max_partition_length(500)
+                .with_max_num_partitions(2)
+                .with_max_partition_contributions(1),
+        )?
+        .with_margin(
+            &["chunk_2_bool", "cycle_5_alpha"],
+            Margin::new()
+                .with_public_keys()
+                .with_max_partition_length(200),
+        )?
+        .with_margin(
+            &["chunk_(..10u32)"],
+            Margin::new()
+                .with_public_keys()
+                .with_max_partition_length(100),
+        )?;
```

Comment:
Are these margins chosen for particular tests? Maybe leave these off, and let the individual tests add margins to the domain?

Reply:
I came up with a combination of margins and dtypes that would cover most use-cases. They get used in specific tests later on. I'm not super keen on breaking their attribution apart across the relevant PRs...